### PR TITLE
Add support for multiple `pMapSkip`'s

### PR DESCRIPTION
**Pull request by ferrinweb**
like this:
```javascript
console.log(await pMap([1, pMapSkip, 2, pMapSkip, 3], async value => value)) // [1, 2, 3]
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
@huntharo Would you be willing to help review?

**Review by huntharo — CHANGES_REQUESTED** [review]
I think we need to address the `O(n^2)` complexity on the number of `pMapSkip` items and fix the existing performance issue with the usage of `.splice()` in a loop.

I left an example of how to test the performance and how to change the `.unshift()` usage back to `.push()` with a loop that does a `.pop()` (using the array as a `O(1)` stack).  The only remaining work would be to implement the `super splice` loop that loops through `result` only a single time and touches each item only a single time.

File: index.js [hunk 1]
```diff
@@ -87,7 +90,7 @@ export default async function pMap(
 					const value = await mapper(element, index);
 
 					if (value === pMapSkip) {
-						skippedIndexes.push(index);
+						skippedIndexes.unshift(index);
```
[the remark this answers is not in the document]
Comment (ferrinweb) [reply index.js]: Thanks!
I made improvements in the latest commit.
Help review, please!

[the remark this answers is not in the document]
Comment (ferrinweb) [reply index.js]: Thanks!
I made improvements in the latest commit.
Help review, please!

File: test.js [hunk 2]
```diff
@@ -155,6 +155,19 @@ test('pMapSkip', async t => {
 	], async value => value), [1, 2]);
 });
 
+test('multiple pMapSkips', async t => {
```
[the remark this answers is not in the document]
Comment (ferrinweb) [reply test.js:158]: Added.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**